test/over_react_test/utils/shadow_nested_component.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:html';
+
+import 'package:over_react/over_react.dart';
+import 'package:over_react/react_dom.dart';
+
+UiFactory<UiProps> ShadowNested = uiFunction(
+  (props) {
+    final divRef = useRef<DivElement>();
+
+    useEffect(() {
+      var inner = DivElement();
+      divRef.current.attachShadow({'mode':'open'}).append(inner);
+      render((Dom.div()
+        ..addUnconsumedDomProps(props, {})
+        ..addTestId('inner')
+      )(), inner);
+    }, [divRef.current]);
```

Comment:
Refs generally shouldn't be used as effect dependencies; in this case, `.current` is null, so this will always rerender on the second render. Also, props should be used as a dependency a) so that the value doesn't become stale after the second render and b) so that this rerenders when props are updated
```dart
    }, [props]);
```
Also, if this component is ever rerendered, the content will be duplicated. Could you update it to reuse the same mount node?

Or, as opposed to those changes, you could just make the effect only run once:
```dart
    }, []);
```

---

Also, since we're using render the nested component will unmount itself, and no unmount call in the effect cleanup is necessary, but that might not be obvious to future maintainers. Could you add a comment?